akvo/iati/exports/elements/result.py

```diff
@@ -17,86 +17,156 @@ def result(project):
-                narrative_element.text = indicator.title
+                narrative_element.text = res.title
 
-            if indicator.description:
-                description_element = etree.SubElement(indicator_element, "description")
+            if res.description:
+                description_element = etree.SubElement(element, "description")
                 narrative_element = etree.SubElement(description_element, "narrative")
                 narrative_element.text = res.description
 
-            if indicator.baseline_year and indicator.baseline_value:
-                baseline_element = etree.SubElement(indicator_element, "baseline")
-                baseline_element.attrib['year'] = str(indicator.baseline_year)
-                baseline_element.attrib['value'] = indicator.baseline_value
-
-                if indicator.baseline_comment:
-                    comment_element = etree.SubElement(baseline_element, "comment")
-                    narrative_element = etree.SubElement(comment_element, "narrative")
-                    narrative_element.text = indicator.baseline_comment
-
-            for period in indicator.periods.all():
-                period_element = etree.SubElement(indicator_element, "period")
-
-                if period.period_start:
-                    period_start_element = etree.SubElement(period_element, "period-start")
-                    period_start_element.attrib['iso-date'] = str(period.period_start)
-
-                if period.period_end:
-                    period_end_element = etree.SubElement(period_element, "period-end")
-                    period_end_element.attrib['iso-date'] = str(period.period_end)
-
-                if period.target_value:
-                    target_element = etree.SubElement(period_element, "target")
-                    target_element.attrib['value'] = period.target_value
-
-                    if period.target_comment:
-                        comment_element = etree.SubElement(target_element, "comment")
-                        narrative_element = etree.SubElement(comment_element, "narrative")
-                        narrative_element.text = period.target_comment
-
-                if period.actual_value:
-                    actual_element = etree.SubElement(period_element, "actual")
-                    actual_element.attrib['value'] = period.actual_value
-
-                    if period.actual_comment:
-                        comment_element = etree.SubElement(actual_element, "comment")
-                        narrative_element = etree.SubElement(comment_element, "narrative")
-                        narrative_element.text = period.actual_comment
-
-                indicator_element.append(period_element)
-
-            element.append(indicator_element)
-
-        result_elements.append(element)
+            for indicator in res.indicators.all():
+                if indicator.measure or indicator.ascending is not None or indicator.title or \
```

Comment:
Now I've seen this kind of really long ifs a number of times and they're hard to read. A helper function along these lines maybe:

```python
def at_least_one_value_field(obj, fields):
    "Returns True if at least one field on obj has a truthy value, otherwise returns False"
    for attr in attrs:
        if getattr(obj, attr, None):
            return True
    return False

fields = ['measure', 'title', 'description', 'baseline_year', 'baseline_value', 'baseline_comment']
if (at_least_one_value_field(indicator, fields) or indicator.ascending is not None
           or indicator.references.all() or indicator.periods.all()):
...
```

Also note that by having the ORM queries last, there's a good chance they never have to be checked and they're the most expensive by far. (Maybe this is too much nit-picking? I can't decide, but those 4-line if statements are ugly!)